### PR TITLE
Improve FilterPill component

### PR DESCRIPTION
**Pull request by LudoVio**
**What is this feature?**

Add a nowrap on the pills

### CODE REVIEW

File: packages/grafana-ui/src/components/FilterPill/FilterPill.tsx [hunk 1]
```diff
@@ -13,13 +13,18 @@ export interface FilterPillProps {
   label: string;
   onClick: React.MouseEventHandler<HTMLElement>;
   icon?: IconName;
+  customClass?: string;
```
Comment (joshhunt) [inline packages/grafana-ui/src/components/FilterPill/FilterPill.tsx:16]: >  set a custom class for further customization from the component using it

This is explicitly an anti-goal for @grafana/ui. "Further customization" means more inconsistency, and more unreliability as implementation details of the components are iterated on.

For your use case, I think it's fine to add `white-space: nowrap` directly to this component. I don't think FilterPill should be used for long content that we want to wrap, and it appears the only use of it is in Transformations. Doesn't look like its used in plugins we know about.

File: packages/grafana-ui/src/components/FilterPill/FilterPill.tsx [hunk 2]
```diff
@@ -48,6 +53,7 @@ const getStyles = (theme: GrafanaTheme2) => {
     selected: css`
       color: ${theme.colors.text.primary};
       background: ${theme.colors.action.selected};
+      border: 1px solid ${theme.colors.warning.main};
```
[the remark this answers is not in the document]
Comment (torkelo) [reply packages/grafana-ui/src/components/FilterPill/FilterPill.tsx:56]: We should probably use primary color (ie blue) here not warning, is there a figma design for this filter pill change?

File: packages/grafana-ui/src/components/FilterPill/FilterPill.tsx (hunk 1 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/FilterPill/FilterPill.tsx:16]: agree, and customClass is also the wrong way to solve it if we where to allow customization (it should be done with className and forwardRef)

File: packages/grafana-ui/src/components/FilterPill/FilterPill.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (torkelo) [reply packages/grafana-ui/src/components/FilterPill/FilterPill.tsx:56]: it's also the wrong color (.main is not designed for border, but the warning.border is)

File: packages/grafana-ui/src/components/FilterPill/FilterPill.tsx (hunk 1 above)
Comment (LudoVio) [reply packages/grafana-ui/src/components/FilterPill/FilterPill.tsx]: removed the customClass

File: packages/grafana-ui/src/components/FilterPill/FilterPill.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (LudoVio) [reply packages/grafana-ui/src/components/FilterPill/FilterPill.tsx:51]: I was following this figma:
https://www.figma.com/file/0WWfKHXFKRt4UI8wtjU8o9/Transformations---Usability-Test---V1---Scenario-2?type=design&node-id=0-1&t=wFUHoGVvXYwUnnbG-0

But there was a problem with the gradient version. There was a talk on slack with SAGA and they seemed to be ok with the orange one.

https://raintank-corp.slack.com/archives/CT7S455SP/p1686328922719439?thread_ts=1686326735.948929&cid=CT7S455SP

Maybe I will wait and discuss that one with our UX then

**Comment by LudoVio** [comment]
So the PR is actually just for adding the nowrap. Will discuss with our UX about the selected state border color. If we do decide to change it, then I will create a proper property in the theme object for selected border state

**Comment by torkelo** [comment]
I think the smallest change to make it a bit better is to change background to secondary.shade (the hover background for secondary button)
Screenshot from 2023-06-19 14-10-12

Alternatively use primary.border 
Screenshot from 2023-06-19 14-07-39

thoughts @staton-hyse11  (see slack thread from josh in #saga-squad for more)

**Comment by staton-hyse11** [comment]
@torkelo if the active state of these tabs go to primary.border, we will need to adjust all active states to be primary.border. Across the product, we are using orange to indicate something is active. It would be good to align all active states.

**Comment by torkelo** [comment]
@staton-hyse11 I don't think we are using orange (or blue) to indicate selected state anywhere. 

We are using orange borders for the toolbar button to indicate "streaming" state in Explore (the only place I know we use this). Also used to indicate activate state for the query history or inspector views in Explore. But this is not a pattern I have seen we use anywhere else. 

I am not proposing we use blue border for all selected states, think the smallest tweak here is probably to use secondary.shade

**Comment by staton-hyse11** [comment]
Here are the active states that I know of within the product:
image
image
image
image


All of these are indicating that something is "active"; that might be streaming as you mentioned but it is still an active or selected state. Plus, the orange border is already present on connections as shown. This would be making that change the component

**Comment by torkelo** [comment]
@staton-hyse11 was not aware of the connections plugin filter pill (not in the design system). 

I was aware of the active state for toolbar button in explore (something that have always looked a bit off / sloppy)

**Comment by torkelo** [comment]
Interesting, so the ToolbarButton active state is using the old theme color "orangeDark", that does not exist the new theme (The tabs are using theme.gradients.brandVertical) which cannot be used for this.

**Comment by catherineymgui** [comment]
Trying to understand what the next steps are based on this thread: https://raintank-corp.slack.com/archives/C04D07D6MDJ/p1686917948219329

Sounds like there's agreement that the orange border can be used. But we will need to create a new color token? @torkelo @staton-hyse11 ?

**Comment by torkelo** [comment]
@catherineymgui think your commenting on the wrong PR (think we should close this one), there is a new PR that tries to fix FilterPill and align how we show activate state. 

 https://github.com/grafana/grafana/pull/70396

> Sounds like there's agreement that the orange border can be used.

Yes, a the gradient orange bottom border we already use in many places (this means we need no new theme token as there already exists on for this). used in Tabs, IconButton and few other places to show active / selected state.

**Pull request closed without merging**